qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/testing/JavaFlow.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.navigation.qa_test_app.testing
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * This is not production tested. It is used to showcase java interoperability.
+ */
+object JavaFlow {
+
+    @JvmStatic
+    fun lifecycleScope(owner: LifecycleOwner): CoroutineScope = owner.lifecycleScope
+
+    @JvmStatic
+    fun <T> collect(flow: Flow<T>, scope: CoroutineScope, consumer: Consumer<T>) {
+        scope.launch {
+            flow.collect { value -> consumer.accept(value) }
+        }
+    }
+}
+
+interface Consumer<T> {
```

Comment:
There is `java.util.function.Consumer`, why do you need this one?

Reply:
Didn't use it because it requires api 24

![Screen Shot 2022-09-12 at 12 52 13 PM](https://user-images.githubusercontent.com/3021882/189743282-268ebc3a-9662-40e1-8c71-cca66dd79530.png)